olp-cpp-sdk-core/src/thread/PriorityQueueExtended.h

```diff
@@ -137,7 +138,7 @@ void PriorityQueueExtended<T, COMPARE>::push(const T& value) {
 
 template <class T, class COMPARE>
 void PriorityQueueExtended<T, COMPARE>::push(T&& value) {
-  impl_->Push(value);
+  impl_->Push(std::forward<T>(value));
```

Comment:
std::forward only works on forwarding references and T is not a forwarding reference as it is specified at class instantiation and not on method call. So here it will have no effect at all, see https://en.cppreference.com/w/cpp/language/reference#Forwarding_references:

```cpp
template<class T> struct A {
    template<class U>
    A(T&& x, U&& y, int* p); // x is not a forwarding reference: T is not a
                             // type template parameter of the constructor,
                             // but y is a forwarding reference
};
```

Reply:
good catch :-)
`T&&` here is an rvalue reference type, but `value` is still variable with name so its an lvalue and must be moved. Thus, `std::forward` is same as `std::move`.
replaced it with `std::move` , accordingly.